File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/ActionConverter.java
package com.platdmit.simplecloudmanager.domain.converters;

import com.platdmit.simplecloudmanager.data.api.models.ApiAction;
import com.platdmit.simplecloudmanager.data.database.entity.DbAction;
import com.platdmit.simplecloudmanager.domain.models.Action;

import java.util.List;

public interface ActionConverter {
    DbAction fromApiToDb(ApiAction apiAction);
    Action fromDbToDomain(DbAction dbAction);
    List<Action> fromDbToDomainList(List<DbAction> dbList);
    List<DbAction> fromApiToDbList(List<ApiAction> apiList);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/ActionConverter.kt
package com.platdmit.simplecloudmanager.domain.converters

import com.platdmit.data.api.models.ApiAction
import com.platdmit.data.database.entity.DbAction
import com.platdmit.simplecloudmanager.domain.models.Action

interface ActionConverter {
    fun fromApiToDb(apiAction: ApiAction): DbAction
    fun fromDbToDomain(dbAction: DbAction): Action
    fun fromDbToDomainList(dbList: List<DbAction>): List<Action>
    fun fromApiToDbList(apiList: List<ApiAction>): List<DbAction>
}